### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR fixes all Pydantic deprecation warnings for models already migrated to v2 but still using methods from v1.

Replace calls to
- `parse_obj` with `model_validate`
- `dict` with `model_dump`
- `Extra.ignore` with `'ignore'`

### Fixed
- Pydantic deprecation warnings

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/apps/demultiplex/sample_sheet/models.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class FlowCellSample(BaseModel):
     sample_id: SampleId
     index: str
     index2: str = ""
-    model_config = ConfigDict(populate_by_name=True, extra=Extra.ignore)
+    model_config = ConfigDict(populate_by_name=True, extra="ignore")
```
Comment (islean) [inline cg/apps/demultiplex/sample_sheet/models.py:24]: Would we want to create a constant for "ignore?

File: cg/server/api.py [hunk 2]
```diff
@@ -489,7 +489,7 @@ def parse_orderform():
         error_message = error.message if hasattr(error, "message") else str(error)
         http_error_response = http.HTTPStatus.INTERNAL_SERVER_ERROR
     else:
-        return jsonify(**parsed_order.dict())
+        return jsonify(**parsed_order.model_dump())
```
Comment (islean) [inline cg/server/api.py:492]: What does jsonify do? Does it do the same as `return parsed_order.model_dump_json()`?

**Review by islean — APPROVED** [review]
Looks good!

File: cg/apps/demultiplex/sample_sheet/models.py (hunk 1 above)
Comment (seallard) [reply cg/apps/demultiplex/sample_sheet/models.py:24]: Maybe if it is used elsewhere, but it is not now so I think it is fine

File: cg/server/api.py (hunk 2 above)
Comment (seallard) [reply cg/server/api.py:492]: What is jsonify?

Comment (seallard) [reply cg/server/api.py:492]: Oh, did not even notice it. Hm, no idea. Reading docs...
```
This function wraps dumps to add a few enhancements that make
life easier. It turns the JSON output into a ~flask.Response
```

Comment (seallard) [reply cg/server/api.py:492]: So I don't think it returns the same as `model_dump_json()`

Comment (islean) [reply cg/server/api.py:492]: Oh, not the same then! model_dump_json() gives a string I think.

Comment (islean) [reply cg/server/api.py:492]: Or maybe? All examples seems to give a string formatted as the class...

Comment (seallard) [reply cg/server/api.py:492]: Might be the case! Going to leave it for now, seems a little out of scope

**Pull request merged**